viz_scripts/mode_specific_timeseries.ipynb

```diff
@@ -0,0 +1,393 @@
+    "active_users.rename(columns={'user_id':'active_users'}, inplace=True)\n",
+    "\n",
+    "# Add 7-day rolling avg smoothing to better see trends\n",
+    "mode_counts['trip_count_smooth'] = mode_counts.groupby(['user_id','Mode_confirm'])['trip_count'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "mode_distance['distance_miles_smooth'] = mode_distance.groupby(['user_id','Mode_confirm'])['distance_miles'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "emissions['distance_miles_smooth'] = emissions.groupby(['user_id'])['distance_miles'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "energy['distance_miles_smooth'] = energy.groupby(['user_id'])['distance_miles'].apply(lambda x: x.rolling(7,1).mean())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Timeseries Plots"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Daily trips for mode of interest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of total ebikeshare trips across all users\n",
+    "plot_data = mode_counts[mode_counts['Mode_confirm']==mode_of_interest]\n",
+    "plot_data = plot_data.groupby(['date_time'])['trip_count_smooth'].agg(['sum']).reset_index()\n",
+    "\n",
+    "plot_title = f'Daily {mode_of_interest} Trips\\n%s'%quality_text\n",
+    "ylab = 'Trip Count'\n",
+    "file_name = \"ts_ebike_share%s\"%file_suffix\n",
+    "timeseries_plot(plot_data['date_time'], plot_data['sum'], plot_title, ylab, file_name)\n",
+    "alt_text = scaffolding.store_alt_text_timeseries(plot_data, file_name, plot_title)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Daily mileage for mode of interest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of total ebikeshare mileage across all users\n",
+    "plot_data = data[data['Mode_confirm']==mode_of_interest]\n",
+    "plot_data = plot_data.groupby(['date_time'])['distance_miles'].agg(['sum']).reset_index()\n",
+    "\n",
+    "plot_title = f'Daily {mode_of_interest} Mileage\\n%s'%quality_text\n",
+    "ylab = 'Miles'\n",
+    "file_name = \"ts_ebike_miles%s\"%file_suffix\n",
+    "timeseries_plot(plot_data['date_time'], plot_data['sum'], plot_title, ylab, file_name)\n",
+    "alt_text = scaffolding.store_alt_text_timeseries(plot_data, file_name, plot_title)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Daily mileage per user for mode of interest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot of total ebikeshare mileage normalized by number of users\n",
+    "plot_data = data[data['Mode_confirm']==mode_of_interest]\n",
```

Comment:
nit: Seems like you can do this once at the beginning of the notebook

Reply:
Consolidated this to occur as few times as possible